front/app/containers/Admin/initiatives/settings/Cosponsors.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+
+import { SectionField, SubSectionTitle } from 'components/admin/Section';
+import { FormattedMessage, useIntl } from 'utils/cl-intl';
+import messages from '../messages';
+import useFeatureFlag from 'hooks/useFeatureFlag';
+import { Box, Toggle, Text, Input } from '@citizenlab/cl2-component-library';
+
+import Error from 'components/UI/Error';
+import errorMessages from 'components/UI/Error/messages';
+
+interface Props {
+  requireCosponsors: boolean | undefined;
+  cosponsorsNumber: number | undefined;
+  onChangeRequireSponsors: (value: boolean) => void;
+  onChangeCosponsorsNumber: (value: number) => void;
+}
+
+const Cosponsors = ({
+  requireCosponsors,
+  cosponsorsNumber,
+  onChangeRequireSponsors,
+  onChangeCosponsorsNumber,
+}: Props) => {
+  const initiativeCosponsorsAllowed = useFeatureFlag({
+    name: 'initiative_cosponsors',
+    onlyCheckAllowed: true,
+  });
+  const { formatMessage } = useIntl();
+
+  if (!initiativeCosponsorsAllowed) return null;
+
+  const handleReactingTresholdOnChange = (value: string) => {
+    onChangeCosponsorsNumber(parseInt(value, 10));
+  };
+
+  return (
+    <SectionField>
+      <SubSectionTitle style={{ marginBottom: '0px' }}>
+        <FormattedMessage {...messages.cosponsors} />
+      </SubSectionTitle>
+      <Box mb={requireCosponsors ? '8px' : '0'}>
+        <Toggle
+          checked={
+            typeof requireCosponsors === 'boolean' ? requireCosponsors : false
+          }
+          onChange={() => {
+            onChangeRequireSponsors(!requireCosponsors);
+          }}
+          label={
+            // copied from front/app/containers/Admin/initiatives/settings/RequireApprovalToggle.tsx
+            <Box ml="8px">
+              <Box display="flex">
+                <Text
+                  color="primary"
+                  mb="0px"
+                  fontSize="m"
+                  style={{ fontWeight: 600 }}
+                >
+                  <FormattedMessage {...messages.requireCosponsorsLabel} />
+                </Text>
+              </Box>
+
+              <Text color="coolGrey600" mt="0px" fontSize="m">
+                <FormattedMessage {...messages.requireCosponsorsInfo} />
+              </Text>
+            </Box>
+          }
+        />
+      </Box>
+      {requireCosponsors && (
+        <>
+          <Box mb="10px">
+            <Input
+              name="cosponsors_number"
+              type="number"
+              min="1"
+              required
+              value={cosponsorsNumber?.toString()}
+              onChange={handleReactingTresholdOnChange}
+              label={formatMessage(messages.cosponsorsNumberLabel)}
+            />
+          </Box>
+          {Number.isNaN(cosponsorsNumber) && (
```

Comment:
Number.isNaN > isNaN. isNaN has unexpected behavior when checking undefined.

`isNaN(undefined) // true`
`Number.isNaN(undefined) // false (what we want)`

Reply:
Oh it's a typo. It should be `isNil(cosponsorsNumber)`.

Reply:
It seems we need both to cover the UI interactions we created:
-Initially, when we first enable the setting, cosponsorsNumber will be undefined.
-If we enable it, fill in a number value, then erase that, parseInt will convert the empty string to NaN (which qualifies as a number)

I tried to minimize this messy part, but it's not so easy if you allow people to type in an input field (with type="number"). If you don't allow erasing of everything (to avoid the conversion to NaN), you get weird behavior in the input box. Anyway, I could explain it in even more detail but added some [comments](https://github.com/CitizenLabDotCo/citizenlab/pull/5471/commits/4fc9ff834efdf5a6902ce23d20777fd681d0ad1d) in the code. Edge case behavior. ;-) Thanks for your comment!